docs/development/javascript/how-to-build-and-use-vuejs-components/index.md

```diff
@@ -2,388 +2,561 @@
-```html
+We've created another component called `rating-title` to illustrate slot nesting. This component will wrap a title that you set inside a pair of `<h1>` tags:
+
+![Components with nested slots](component-nested-slots.png "Components with nested slots")
+
+Let's explore the code for this component:
+
+-   The template for the new component is defined on lines 37-41. The `<slot>` tag has been added to this template, but this time the slot is not empty.
+
+-   You can specify the default value for slots by adding it between the open and close slot tags: ```<slot>Default Rating Title</slot>```
+
+-   You can see how this default value is referenced for the new `rating-title` component on line 16. Whenever nothing is included between the `<rating-title></rating-title>` tags, the default value is used.
+
+-   Compare this with how the component is used on lines 8 and 12. Because a title is included between the component's tags, the default text is is not rendered.
+
+### Named Slots
+
+To allow even more structure, you can use multiple slots in a template by using [slot naming](https://vuejs.org/v2/guide/components-slots.html#Named-Slots). Lets overengineer our simple example a little bit to see how it works.
+
+Replace the contents of your `ratingcounter.html` with the following snippet:
+
+{{< file "ratingcounter.html" html >}}
 <script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
 
 <div id="app">
-   <rating-counter v-bind:parent="parentTitle">
+    {{ parentHeader.label }}{{ parentHeader.globalCount }}
+    <hr />
+
+   <rating-counter v-bind:parent="parentHeader">
        <template v-slot:title>
            <rating-title></rating-title>
        </template>
        <template v-slot:subtitle>
-           <h2>Subtitle</h2>
+           Subtitle
        </template>
    </rating-counter>
 </div>
+
 <script>
-   Vue.component('rating-counter', {
-       props: ['parent'],
-       data() {
-           return {
-               count: 0
-           }
-       },
-       template:   `<div>
-                       <slot name="title"></slot>
-                       <slot name="subtitle"></slot>
-                       <button v-on:click="count--;parent.counter--;">Thumbs Down</button>
-                       {{ count }}
-                       <button v-on:click="count++;parent.counter++;">Thumbs up</button>
-                   </div>`
-   })
-
-   Vue.component('rating-title', {
-       template:   `<div>
-                       <slot><h1>Hello rating title</h1></slot>
-                   </div>`
-   })
-    new Vue({
-       el: '#app',
-       data: {
-           parentTitle: {
-               title: "Counter is at: ",
-               counter: 0
-           }
-       }
-   })
+Vue.component('rating-counter', {
+    props: ['parent'],
+    data() {
+        return {
+            count: 0
+        }
+    },
+    template:   `<div>
+                    <slot name="title"></slot>
+                    <h2>
```

Comment:
Moved the h2 tags for the subtitle out of the main HTML and into the template for the rating-counter component.
- This could help better illustrate that the subtitle slot will embed the content wherever it appears inside the template.
- In the previous version of this example, the title and subtitle slots appear right next to each other both in the main HTML and in the template, so there wasn't any sort of extra styling added by the slot placement in the template.